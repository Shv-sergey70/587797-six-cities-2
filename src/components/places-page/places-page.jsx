import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {Map} from "../map/map";
import CitiesList, {cityPropTypes} from "../cities-list/cities-list";
import {offerPropTypes} from "../card/card";
import Selectors from "../../selector";
import {withActiveItem} from "../../hocs/with-active-item/with-active-item";
import {OffersList} from "../offers-list/offers-list";

const CitiesListWrapped = withActiveItem(CitiesList);
const OffersListWrapped = withActiveItem(OffersList);

const PlacesPage = (props) => {
  const {
    currentCity,
    currentOffers
  } = props;

  return <main className="page__main page__main--index">
    <h1 className="visually-hidden">Cities</h1>
    <div className="tabs">
      <section className="locations container">
        <CitiesListWrapped
          defaultActiveItem={currentCity}
        />
      </section>
    </div>
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">{currentOffers.length} places to stay in {currentCity.name}</b>
          <form className="places__sorting" action="#" method="get">
            <span className="places__sorting-caption">Sort by</span>
            <span className="places__sorting-type" tabIndex="0">
              Popular
              <svg className="places__sorting-arrow" width="7" height="4">
                <use xlinkHref="#icon-arrow-select"> </use>
              </svg>
            </span>
            <ul className="places__options places__options--custom places__options--opened">
              <li className="places__option places__option--active" tabIndex="0">Popular</li>
              <li className="places__option" tabIndex="0">Price: low to high</li>
              <li className="places__option" tabIndex="0">Price: high to low</li>
              <li className="places__option" tabIndex="0">Top rated first</li>
            </ul>
          </form>
          <OffersListWrapped
            offers={currentOffers}
            defaultActiveItem={0}
          />
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <Map
              offersLocations={currentOffers.map((offer) => offer.location)}
            />
          </section>
        </div>
      </div>
    </div>
  </main>;
};

PlacesPage.propTypes = {
  currentCity: cityPropTypes,
  currentOffers: PropTypes.arrayOf(offerPropTypes)
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: state.currentCity,
  currentOffers: Selectors.getCurrentOffers(state)
});

export {PlacesPage};

export default connect(mapStateToProps)(PlacesPage);
