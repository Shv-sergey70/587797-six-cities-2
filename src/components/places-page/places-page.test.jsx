import React from 'react';
import renderer from 'react-test-renderer';
import {PlacesPage} from "./places-page";

jest.mock(`../cities-list/cities-list.jsx`, () => `CitiesList`);
jest.mock(`../sort/sort.jsx`, () => `Sort`);
jest.mock(`../offers-list/offers-list.jsx`, () => `OffersList`);
jest.mock(`../map/map.jsx`, () => `Map`);

const offersData = [
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    type: `apartment`,
    price: 80,
    rating: 4.4,
    bedrooms: 2,
    maxAdults: 2,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/studio-01.jpg`],
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    description: `A quiet cozy and picturesque that hides behind a a river`,
    location: {
      latitude: 123,
      longitude: 456,
      zoom: 10
    }
  },
  {
    id: 2,
    title: `Beautiful & luxurious apartment at great location2`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: false,
    isFavorite: true,
    type: `apartment`,
    price: 180,
    rating: 2.4,
    bedrooms: 5,
    maxAdults: 3,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/studio-01.jpg`],
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`],
    description: `A quiet cozy and picturesque that hides behind a a river1`,
    location: {
      latitude: 123,
      longitude: 456,
      zoom: 10
    }
  }
];

it(`Places page correctly renders`, () => {
  const tree = renderer.create(
      <PlacesPage
        currentCity = {{
          name: `Paris`,
          location: {
            latitude: 11,
            longitude: 22,
            zoom: 10
          }
        }}
        currentOffers={offersData}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
