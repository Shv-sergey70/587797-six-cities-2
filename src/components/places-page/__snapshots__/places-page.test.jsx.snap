// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places page correctly renders 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <CitiesList />
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Paris
        </b>
        <Sort
          isOpen={false}
          openStateChange={[Function]}
        />
        <OffersList
          offers={
            Array [
              Object {
                "bedrooms": 2,
                "description": "A quiet cozy and picturesque that hides behind a a river",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "id": 1,
                "images": Array [
                  "/img/room.jpg",
                  "/img/apartment-01.jpg",
                  "/img/studio-01.jpg",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 123,
                  "longitude": 456,
                  "zoom": 10,
                },
                "maxAdults": 2,
                "previewImage": "/img/apartment-01.jpg",
                "price": 80,
                "rating": 4.4,
                "title": "Beautiful & luxurious apartment at great location",
                "type": "apartment",
              },
              Object {
                "bedrooms": 5,
                "description": "A quiet cozy and picturesque that hides behind a a river1",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                ],
                "id": 2,
                "images": Array [
                  "/img/room.jpg",
                  "/img/apartment-01.jpg",
                  "/img/studio-01.jpg",
                ],
                "isFavorite": true,
                "isPremium": false,
                "location": Object {
                  "latitude": 123,
                  "longitude": 456,
                  "zoom": 10,
                },
                "maxAdults": 3,
                "previewImage": "/img/apartment-01.jpg",
                "price": 180,
                "rating": 2.4,
                "title": "Beautiful & luxurious apartment at great location2",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <Map
            offersLocations={
              Array [
                Object {
                  "latitude": 123,
                  "longitude": 456,
                  "zoom": 10,
                },
                Object {
                  "latitude": 123,
                  "longitude": 456,
                  "zoom": 10,
                },
              ]
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
